OvmfPkg/Include/Dsc/OvmfPkg.dsc.inc

```diff
@@ -2,6 +2,10 @@
 #  SPDX-License-Identifier: BSD-2-Clause-Patent
 ##
 
+# ASM only SEC files cannot link against this in MSVC as it tries to link the ModuleEntryPoint, which doesn't exist
+[LibraryClasses.common.PEI_CORE, LibraryClasses.common.PEIM, LibraryClasses.common.DXE_CORE, LibraryClasses.common.SMM_CORE, LibraryClasses.common.DXE_DRIVER, LibraryClasses.common.DXE_RUNTIME_DRIVER, LibraryClasses.common.DXE_SMM_DRIVER, LibraryClasses.common.MM_STANDALONE, LibraryClasses.common.UEFI_DRIVER, LibraryClasses.common.UEFI_APPLICATION]
+  NULL|MdePkg/Library/StackCheckLibNull/StackCheckLibNull.inf
+

```

Comment:
This is kind of unfortunate, and IIUC, this is what is preventing us from adding the NULL resolution generally to the include file in MdePkg.

So why can we not add the kludge above to MdeLibs.inc instead?

Reply:
Agreed it is unfortunate. We could add the kludge to MdeLibs.inc instead, if we can say that only SEC will have components that do not rely on `__ModuleEntryPoint`. We would still have to touch every DSC to add the SEC variety (either as a whole or for SEC components that do have entry points).

Reply:
I'm going to run a few tests here and see if I can figure out what is adding that dependency, because it shouldn't be there, then we could go with the simple case.

Reply:
It looks like we can say that: https://github.com/tianocore/edk2/blob/be1d408773504df8b95a2a81be7b558276ba5356/BaseTools/Source/Python/AutoGen/GenC.py#L1460-L1462

HOST_APPLICATION is already taken care of in this set by disabling stack cookies for them and USER_DEFINED is by definition in the hands of the user :). I will make this change.